lms/djangoapps/teams/csv.py

```diff
@@ -19,3 +28,217 @@ def load_team_membership_csv(course, response):
+            self.teamset_name_by_index[i] = header_row[i]
+        return True
+
+    def validate_user_entry(self, user):
+        """
+        Invalid states:
+            user not enrolled in course
+        Validates user row entry. Returns true if there are no errors.
+        user_row is the list representation of an input row. It will have the following formta:
+        use_id, enrollment_mode, <Team_Name_1>,...,<Team_Name_n>
+        Team_Name_x are optional and can be a sparse list i.e:
+        andrew,masters,team1,,team3
+        joe,masters,,team2,team3
+        """
+        if not CourseEnrollment.is_enrolled(user, self.course.id):
+            self.validation_errors.append('User ' + user.username + ' is not enrolled in this course.')
+            return False
+
+        return True
+
+    def validate_user_to_team(self, user_row):
+        """
+        Validates a user entry relative to an existing team.
+        user_row is the list representation of an input row. It will have the following formta:
+        user_row[0] will contain an edX user object, followed by:
+        enrollment_mode, <Team_Name_1>,...,<Team_Name_n>
+        Team_Name_x are optional and can be a sparse list i.e:
+        [andrew],masters,team1,,team3
+        [joe],masters,,team2,team3
+        """
+        for i in range(2, len(user_row)):
+            user = user_row[0]
+            team_name = user_row[i]
+            if team_name:
+                try:
+                    # checks for a team inside a specific team set. This way team names can be duplicated across
+                    # teamsets
+                    team = CourseTeam.objects.get(name=team_name, topic_id=self.teamset_name_by_index[i])
+                except CourseTeam.DoesNotExist:
+                    # if a team doesn't exists, the validation doesn't apply to it.
+                    if user.id in self.user_ids_by_teamset_index[i] and self.add_error_and_check_if_max_exceeded(
+                            'User ' + user.id.__str__() + ' is already on a team set.'):
+                        return False
+                    else:
+                        self.user_ids_by_teamset_index[i].add(user.id)
+                        continue
+                max_team_size = self.course_module.teams_configuration.default_max_team_size
+                if max_team_size is not None and team.users.count() >= max_team_size:
+                    if self.add_error_and_check_if_max_exceeded('Team ' + team.team_id + ' is already full.'):
+                        return False
+                if CourseTeamMembership.user_in_team_for_course(user, self.course.id, team.topic_id):
+                    if self.add_error_and_check_if_max_exceeded(
+                        'The user ' + user.username + ' is already a member of a team inside teamset '
+                        + team.topic_id + ' in this course.'
```

Comment:
same readability idea in here about plucking error messages out into variables and then invoking that method on them.